sparse_strips/vello_common/src/encode.rs

```diff
@@ -0,0 +1,819 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Paints for drawing shapes.
+
+use crate::color::Srgb;
+use crate::color::palette::css::BLACK;
+use crate::kurbo::{Affine, Point, Vec2};
+use crate::peniko::{ColorStop, Extend, GradientKind};
+use smallvec::SmallVec;
+use std::borrow::Cow;
+use std::f32::consts::PI;
+use std::iter;
+use vello_api::paint::{Gradient, IndexedPaint, Paint};
+
+const DEGENERATE_THRESHOLD: f32 = 1.0e-6;
+const NUDGE_VAL: f32 = 1.0e-7;
+
+/// A trait for encoding gradients.
+pub trait Encode {
```

Comment:
Can we call this `EncodeExt` or `GradientExt`? I'd also consider making it sealed.